### PR TITLE
glossary: import terms from nixionary.org

### PR DESCRIPTION
**Pull request by zimbatm**
Nix is using a bunch of esoteric terms like "closure" and "realization"
that can turn off newcomers. I'm assuming that having a good glossary
can help with that. I always like a good glossary to speed-run a new
project.

<https://nixionary.org> was a little project I had for NixCon, to
collect more of those words for the glossary.

### CODE REVIEW

**Comment by yukiisbored** [comment]
@zimbatm Yes, currently @domenkozar still holds the copyright for now.

This will be reassigned to the foundation in the future.

File: source/glossary.md [hunk 1]
```diff
@@ -1,35 +1,95 @@
...
+    A git repository containing a collection of 80000+ packages. See
+    <https://nixpkgs.org>.
 
     Read /nɪks ˈpækɪʤɪz/ ("Nix packages").
 
 NixOS
-    Linux distribution based on Nix and Nixpkgs.
+    The name represents two things at the same time:
+
+    1. The whole project and community.
```
Comment (fricklerhandwerk) [inline source/glossary.md:74]: That's not true and a reason for large amounts of confusion.

File: source/glossary.md [hunk 2]
```diff
@@ -1,35 +1,95 @@
...
+    Per user GC roots. Those are typically set when using `nix-env` or
+    `nixos-rebuild`.
+
 Nix file
     File (`.nix`) containing a Nix expression.
 
 Nixpkgs
-    Software distribution built with Nix.
+    A git repository containing a collection of 80000+ packages. See
+    <https://nixpkgs.org>.
```
Comment (fricklerhandwerk) [inline source/glossary.md:67]: Adding a link is good, more descriptions, I think, less so. Each project should just describe itself authoritatively, as a single source of truth. That description may change over time, no point tracking these things here.

File: source/glossary.md [hunk 3]
```diff
@@ -1,35 +1,95 @@
...
+
+Channels
+    Represents two things:
+    1. Releases of nixpkgs built and pushed by Hydra. See
+       <https://status.nixos.org>.
+    2. GC roots that are managed by `nix-channel`.
+
+Closure
+    The closure is the tree of the dependencies of a derivation, and the
+    dependencies of the depencencies recursively.
```
Comment (fricklerhandwerk) [inline source/glossary.md:16]: These should go to the Nix manual's glossary.

File: source/glossary.md (hunk 1 above)
Comment (zimbatm) [reply source/glossary.md:74]: What is the name of the project and community then? The entry point domain is nixos.org.

I agree that it's confusing, that's why it's worth documenting it.

**Comment by zimbatm** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by fricklerhandwerk** [comment]
I'm trying to add definitions for a part of these terms to the Nix manual:

https://github.com/NixOS/nix/pull/7318
https://github.com/NixOS/nix/pull/7332

[Substituter](https://nixos.org/manual/nix/unstable/glossary.html?highlight=substituter#gloss-substituter) already exists in the Nix manual' glossary.

I'm not sure about the change overall. It's probably good to have multiple entry points for terms. But maybe it's enough to link to the terms' authoritative definitions (in their respective manuals) when they are used in texts.

In any case we should not re-make the definition here, and rather link to the existing one. Sometimes we don't have (a good) one, like it was the case for the Nix language in the Nix manual, but that should be addressed first then.

**Comment by yukiisbored** [comment]
I don't think it's a bad idea to have another glossary in nix.dev since I imagine they rarely change.

Though, I'm not opposed to linking the definition on the Nix manual.

**Comment by zimbatm** [comment]
I'm happy if you incorporate my work in any way that you think is useful. If you want to push all the definitions to the respective projects and then transform this glossary into a meta glossary, that's fine.

Since you seem to be opinionated about it @fricklerhandwerk, I think the easiest would be for you to cannibalize my PR. That way we avoid too much back and forth.

**Comment by fricklerhandwerk** [comment]
@zimbatm thanks, I appreciate the addition in general. Sorry, the tone here is usually quite to the point, so my comment was not to dismiss your effort. This is a strategic decision for the documentation team to make, I just have random opinions which are surely most often wrong.

**Comment by zimbatm** [comment]
No worries, it makes sense to be opinionated for documentation, to give it a consistent feel.

Please have a chat with the documentation team and let me know once you agree on the high-level direction.

**Comment by zimbatm** [comment]
oops, deleted my fork by mistake

**Comment by fricklerhandwerk** [comment]
@NixOS/documentation-team

**Comment by infinisil** [comment]
Imo the nix.dev glossary should ideally be the combined glossaries of [Nix itself](https://nixos.org/manual/nix/stable/glossary.html), nixpkgs and NixOS, plus all of the general terms not mentioned anywhere else (like Nixpkgs, NixOS, etc.)

I'm not sure how feasible this is though

**Comment by zimbatm** [comment]
Imagine all the people who could have benefited from that Glossary, even if imperfect. It would be better than the current situation. And you can still move around later?

**Pull request closed without merging**